packages/anoncreds/tests/InMemoryAnonCredsRegistry.ts

```diff
@@ -94,21 +98,23 @@ export class InMemoryAnonCredsRegistry implements AnonCredsRegistry {
     agentContext: AgentContext,
     options: RegisterSchemaOptions
   ): Promise<RegisterSchemaReturn> {
-    const { namespaceIdentifier, namespace } = parseIndyDid(options.schema.issuerId)
-    const didIndySchemaId = getDidIndySchemaId(
-      namespace,
-      namespaceIdentifier,
-      options.schema.name,
-      options.schema.version
-    )
-    const legacySchemaId = getLegacySchemaId(namespaceIdentifier, options.schema.name, options.schema.version)
+    let legacyIssuerId
+    let didIndySchemaId = ''
+    if (this.useLegacyIdentifiers) {
+      legacyIssuerId = options.schema.issuerId
+    } else {
+      const { namespace, namespaceIdentifier } = parseIndyDid(options.schema.issuerId)
+      legacyIssuerId = namespaceIdentifier
+      didIndySchemaId = getDidIndySchemaId(namespace, namespaceIdentifier, options.schema.name, options.schema.version)
+      this.schemas[didIndySchemaId] = options.schema
+    }
 
+    const legacySchemaId = getLegacySchemaId(legacyIssuerId, options.schema.name, options.schema.version)
```

Comment:
The old implementation supported both for convenience, why is this needed?

Reply:
If schema issuerId is an unqualified did, `parseIndyDid` threw an error. Maybe I can do it 'automatic' (i.e. without config flag and a try-catch?)

Reply:
It's for tests, so whatever works
Is fine. Just curious